lib/ansible/modules/remote_management/redfish/redfish_facts.py

```diff
@@ -0,0 +1,251 @@
+        result = rf_utils._find_systems_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in cmd_list:
+            if command == "GetSystemInventory":
+                result["system"] = rf_utils.get_system_inventory()
+            elif command == "GetPsuInventory":
+                result["psu"] = rf_utils.get_psu_inventory()
+            elif command == "GetCpuInventory":
+                result["cpu"] = rf_utils.get_cpu_inventory()
+            elif command == "GetNicInventory":
+                result["nic"] = rf_utils.get_nic_inventory()
+            elif command == "GetStorageControllerInventory":
+                result["storage_controller"] = rf_utils.get_storage_controller_inventory()
+            elif command == "GetDiskInventory":
+                result["disk"] = rf_utils.get_disk_inventory()
+            elif command == "GetBiosAttributes":
+                result["bios_attribute"] = rf_utils.get_bios_attributes()
+            elif command == "GetBiosBootOrder":
+                result["bios_boot_order"] = rf_utils.get_bios_boot_order()
+
+    elif category == "Chassis":
+        # execute only if we find Chassis resource
+        result = rf_utils._find_chassis_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in cmd_list:
+            if command == "GetFanInventory":
+                result["fan"] = rf_utils.get_fan_inventory()
+
+    elif category == "Accounts":
+        # execute only if we find an Account service resource
+        result = rf_utils._find_accountservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in cmd_list:
+            if command == "ListUsers":
+                result["user"] = rf_utils.list_users()
+
+    elif category == "Update":
+        # execute only if we find UpdateService resources
+        result = rf_utils._find_updateservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in cmd_list:
+            if command == "GetFirmwareInventory":
+                result["firmware"] = rf_utils.get_firmware_inventory()
+
+    elif category == "Manager":
+        # execute only if we find a Manager service resource
+        result = rf_utils._find_managers_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in cmd_list:
+            if command == "GetManagerAttributes":
+                result["mgr_attr"] = rf_utils.get_manager_attributes()
+            elif command == "GetLogs":
+                result["log"] = rf_utils.get_logs()
+
+    # Return data back or fail with proper message
+    if result['ret'] is True:
+        del result['ret']
+        module.exit_json(ansible_facts=result)
```

Comment:
I'm not sure what the contents of `result` look like. It's probably a good idea to put these inside their own key, e.g., `redfish_facts`, rather than have them be "top level" facts.